### PR TITLE
WICKET-6774: minor change to contract of Component.getBehaviorId

### PR DESCRIPTION
**Pull request by papegaaij**
This PR suggest a minor change to the contract of IRequestComponent.getBehaviorId to prepare for a refactoring of the component internals.

### CODE REVIEW

**Comment by papegaaij** [comment]
> Please do not merge this PR until we are clear on the way this is supposed to work.
> https://markmail.org/message/p4cnvfypzprrl2qt

Agreed, that's why I created a PR for such a simple change. The change is simple, but the implications of doing it wrong are much greater. We need have a clear understanding of how we want these stable id's to work.

File: wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java [hunk 1]
```diff
@@ -83,12 +83,12 @@ protected void onBind()
...
 		component.setOutputMarkupId(true);
-		
-		if (getStatelessHint(component))
-		{
-			//generate behavior id
-			component.getBehaviorId(this);
-		}
+	}
+	
+	@Override
```
Comment (bitstorm) [inline wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java:88]: Hi,
forcing Ajax behavior to be stateful by default will break the current logic for stateless Ajax component. See [user guide](https://ci.apache.org/projects/wicket/guide/8.x/single.html#_stateless_ajax_components_behaviors). 
I would remove this isStableIdRequired override and just keep things as they are now, i.e. behaviors state (stateful/stateless) depends on container component.

Comment (papegaaij) [reply wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java:88]: `isStableIdRequired` does not make the behavior stateful, that's the purpose of this new method.

Comment (bitstorm) [reply wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java:88]: ok. thanks for the explanation!

**Pull request merged**